### DISCUSSION TITLE
Prevent code to be re-executed with auto-reload enabled

### DISCUSSION THREAD

**Discussion by AndreuCodina** [fastapi/fastapi-cli#58] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
# application_settings.py
from pydantic_settings import BaseSettings
class ApplicationSettings(BaseSettings):
  field: str


# dependency_container.py
class DependencyContainer:
  @classmethod
  def initialize(cls) -> None:
    self.application_settings = ApplicationSettings(field="value")


# main.py
DependencyContainer.initialize()
app = FastAPI(title="Demo", version="1.0.0")
```

### Description

We read the configurations and secrets from Azure Key Vault (https://github.com/pydantic/pydantic-settings/blob/main/docs/index.md#azure-key-vault) and Azure App Configurations (https://github.com/AndreuCodina/pydantic-settings-azure-app-configuration) when FastAPI is executed with auto-reload and we store them in a `BaseSettings` class (from the `pydantic-settings` package). In summary, we instantiate a singleton of that class.

When a piece of code is changed, the auto-reload executes `main.py` again and the singleton is re-instantiated. Each reading costs money and takes time, so we need to not re-execute the singleton instantiation when FastAPI detects a change.

### Operating System

Linux, Windows, macOS

### Operating System Details

Latest.

### Python Version

3.12.4

**Comment by j-carson** [comment]
If that line is not re-executed, then your singleton won't be initialized at all. 

If you can test and develop some functions without that singleton, I suggest changing your class to not execute the costly code until the first access of azure services is needed. You could also control this by having separate dev and production development environments where developers do not need to go to azure every time they save a code change.

**Comment by YuriiMotov** [comment]
@AndreuCodina, have you found a solution?
Is the answer above suitable for you?

**Reply by AndreuCodina** [reply]
Nope

**Reply by YuriiMotov** [reply]
I think you can try asking this on [`uvicorn` repo](https://github.com/Kludex/uvicorn/discussions). If there is a solution to achieve this with Uvicorn, it should be easily portable to `fastapi-cli`
